crates/compiler/mono/src/ir.rs

```diff
@@ -4245,7 +4249,111 @@ pub fn with_hole<'a>(
             }
         }
 
-        Tuple { .. } => todo!("implement tuple hole"),
+        Tuple {
```

Comment:
Doesn't need to happen now, but we should consolidate the logic in this branch with that of the record branch since they're largely the same.

Reply:
The here is subtly-different-enough that's it's really annoying to combine. I've already taken a few runs at it, unsuccessfully. I propose punting this for now.